File: xample/Xample.DeadCode.CSharp/ClassWithConstructors.cs
namespace Xample.DeadCode.CSharp
{
	public class ClassWithConstructors
	{
		public ClassWithConstructors(EmptyClass cls)
		{
			Empty = cls;
		}

		public ClassWithConstructors(EmptyClass cls, int number)
		{
			Empty = number % 2 == 1 ? cls : null;
		}

		public EmptyClass Empty { get; set; }
	}
}

File: xample/Xample.DeadCode.VisualBasic/ClassWithConstructors.vb
Public Class ClassWithConstructors

	Public Sub New(empty As EmptyClass)
		Me.Empty = empty
	End Sub

	Public Sub New(empty As EmptyClass, number As Integer)
		Me.Empty = If(number Mod 2 = 1, empty, Nothing)
	End Sub

	Public Property Empty As EmptyClass

End Class
